plugins/languages/texpresso.nix

```
@@ -0,0 +1,16 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+helpers.vim-plugin.mkVimPlugin config {
+  name = "texpresso";
+  originalName = "texpresso.vim";
+  defaultPackage = pkgs.vimPlugins.texpresso-vim;
+
+  maintainers = with lib.maintainers; [nickhu];
+
+  extraPackages = with pkgs; [texpresso];
```

```suggestion
  extraOptions = {
    texpressoPackage = mkOption {
        type = with types; nullOr package;
        default = pkgs.texpresso;
        description = ''
          The fzf package to use.
          Set to `null` to not install any package.
        '';
      };
  };
  
  extraPackages = [cfg.texpressoPackage];
```

@MattSturgeon @traxys should we do this systematically or here it is fine to just auto-install `texpresso` ? I have no strong opinion.
It would be nice to agree on a general rule: either
A) Systematically add a `fooPackage` option that can be set to `null`
OR
B) If it makes some sense for a user to select a custom package for it, add the option, otherwise, silent auto-install is OK (more subjective)

On the one hand (A) seems more straightforward to enforce and requires less discussion. It also maximizes the user's control over its setup.
On the other hand, adding `pkgs.git` to `extraPackages` for `plugins.gitsigns` doesn't shock me either.

---

I prefer option "A"

---

Sure, but a user could want nixvim not to install texpresso automatically by setting this option to `null`.
For instance, you could have another version in your `$PATH`.
Also, you could override the `texpresso` derivation right here.
Anyway, my point is that it does no harm having the option.

---

> I would like in the future to have _all_ packages installed by nixvim be nullable to support externally managed programs

So be it !
@NickHu please implement the proposed snippet to add `texpressoPackage`.

---

> and there is no alternative package that would work

A user may wish to use a custom package (e.g. patched), a package from another channel (e.g. stable vs unstable) or some third party flake.

In addition to the null option discussed above.